eq-author/src/App/MetadataModal/MetadataTable/Controls/KeySelect.js

```diff
@@ -34,9 +34,14 @@ class KeySelect extends Component {
   };
 
   getChangeValue = changes => {
+    const regex = /^[a-z0-9-_]+$/i;
     if (changes.hasOwnProperty("selectedItem")) {
       return changes.selectedItem.value;
-    } else if (changes.hasOwnProperty("inputValue")) {
+    } else if (
+      (changes.hasOwnProperty("inputValue") &&
+        regex.test(changes.inputValue)) ||
+      !changes.inputValue
```

Comment:
This feels a bit weird. In the case of `inputValue` being `undefined`, the `else if` will still evaluate to true so we return `undefined`?